lib/std/c.zig

```diff
@@ -295,12 +295,25 @@ pub extern "c" fn kevent(
 ) c_int;
 
 pub extern "c" fn getaddrinfo(
-    noalias node: [*:0]const u8,
-    noalias service: [*:0]const u8,
-    noalias hints: *const addrinfo,
+    noalias node: ?[*:0]const u8,
+    noalias service: ?[*:0]const u8,
+    noalias hints: ?*const addrinfo,
     noalias res: **addrinfo,
 ) EAI;
 
+test "getaddrinfo" {
+    if (!std.builtin.link_libc) return error.SkipZigTest;
+    const allocator = std.testing.allocator;
+    const port = "9999\x00";
+
+    var res: *std.os.addrinfo = undefined;
+    defer freeaddrinfo(res);
```

Comment:
Usually you would put the `defer` after the allocating call.

However, in this case, there is no need for a `defer` at all as you don't use the result. just call `freeaddrinfo` after `getaddrinfo`.